src/events/http/authJWTSettingsExtractor.js

```diff
@@ -0,0 +1,70 @@
+import serverlessLog from '../../serverlessLog.js'
+
+export default function authJWTSettingsExtractor(
+  endpoint,
+  provider,
+  ignoreJWTSignature,
+) {
+  const buildFailureResult = (warningMessage) => {
+    serverlessLog(warningMessage)
+
+    return { unsupportedAuth: true }
+  }
+
+  const buildSuccessResult = (authorizerName) => ({ authorizerName })
+
+  const { authorizer } = endpoint
+
+  if (!authorizer) {
+    return buildSuccessResult(null)
+  }
+
+  if (!provider.httpApi || !provider.httpApi.authorizers) {
+    return buildSuccessResult(null)
+  }
+
+  // TODO: add code that will actually validate a JWT.
+  if (!ignoreJWTSignature) {
+    return buildSuccessResult(null)
```

Comment:
would it be better to throw an error or return failure result if the user specifies that the signature should not be ignored?
They may be unaware that the signature isn't being validated

Reply:
If you don't set `ignoreJWTSignature` you will have to set `noAuth` for serverless-offline to run at all for HttpApi events with authorization. The current behavior of serverless-offline is to treat authorization as another lambda. Since there isn't a function mapped to the authorizer, serverless-offline throws an error on startup. My intention with this flag is to make it explicit that while the authorization rules defined in the serverless.yml file are being tested, this is should not be considered valid security.

Reply:
let me know if you'd like to see the change you suggested. my intention was not to modify the current behavior of the library.